docs/spec_draft.md

```diff
@@ -1586,6 +1587,59 @@ specification. Numeric precision is implementation-defined.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.select
+
+### Semantics
+Produces a `result` tensor where each element is selected from `on_true` or
+`on_false` tensor based on the value of the corresponding element of `pred`.
+More formally, if `predicate[i0, ..., iR-1] = true` then
+`result[i0, ..., iR-1] = on_true[i0, ..., iR-1]`  else
+`result[i0, ..., iR-1] = on_false[i0, ..., iR-1]`, where
+`predicate[i0, ..., iR-1] = pred` if `rank(pred)` $=$ `0` else
+`predicate[i0, ..., iR-1] = pred[i0, ..., iR-1]`.
+
+### Inputs
+
+| Name       | Type                         |
+|------------|------------------------------|
+| `pred`     | tensor of type `i1`          |
+| `on_true`  | tensor of any supported type |
+| `on_false` | tensor of any supported type |
+
+### Outputs
+
+| Name     | Type                         |
+|----------|------------------------------|
+| `result` | tensor of any supported type |
+
+### Constraints
+
+  * (C1) Either `rank(pred)` $=$ `0` or `shape(pred)` $=$ `shape(on_true)`.
+  * (C2) `on_true` and `on_false` have same type.
```

Comment:
`on_true` and `on_false` need to have compatible element types and lower precision type is casted to higher precision type, but I have written they need to be of same type for now.